pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_reconcile.go

```diff
@@ -72,27 +77,120 @@ func (r *endpointsReconciler) reconcile(ctx context.Context, apiExportEndpointSl
 				apisv1alpha1.APIExportValid,
 				apisv1alpha1.APIExportNotFoundReason,
 				conditionsv1alpha1.ConditionSeverityError,
-				"APIExport %s|%s not found",
+				"Error getting APIExport %s|%s",
 				apiExportPath,
 				apiExportEndpointSlice.Spec.APIExport.Name,
 			)
+			conditions.MarkFalse(
+				apiExportEndpointSlice,
+				apisv1alpha1.APIExportEndpointSliceURLsReady,
+				apisv1alpha1.ErrorGeneratingURLsReason,
+				conditionsv1alpha1.ConditionSeverityError,
+				"",
+			)
+			// No need to try again
 			return nil
 		} else {
 			conditions.MarkFalse(
 				apiExportEndpointSlice,
 				apisv1alpha1.APIExportValid,
-				apisv1alpha1.InternalErrorReason,
+				reason,
 				conditionsv1alpha1.ConditionSeverityError,
 				"Error getting APIExport %s|%s",
 				apiExportPath,
 				apiExportEndpointSlice.Spec.APIExport.Name,
 			)
+			conditions.MarkUnknown(
+				apiExportEndpointSlice,
+				apisv1alpha1.APIExportEndpointSliceURLsReady,
+				apisv1alpha1.ErrorGeneratingURLsReason,
+				"",
+			)
 			return err
 		}
 	}
 	conditions.MarkTrue(apiExportEndpointSlice, apisv1alpha1.APIExportValid)
 
-	if err = r.updateEndpoints(ctx, apiExportEndpointSlice, apiExport); err != nil {
+	// Get selector from Partition
+	var selector labels.Selector
+	if apiExportEndpointSlice.Spec.Partition != "" {
+		partition, err := r.getPartition(logicalcluster.From(apiExportEndpointSlice), apiExportEndpointSlice.Spec.Partition)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Don't keep the endpoints if the Partition has been deleted
+				// and is still referenced
+				apiExportEndpointSlice.Status.APIExportEndpoints = nil
+				conditions.MarkFalse(
+					apiExportEndpointSlice,
+					apisv1alpha1.PartitionValid,
+					apisv1alpha1.PartitionInvalidReferenceReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					err.Error(),
```

Comment:
I don't like exposing this error. Rule of thumb: we must not give a hint about existence of a workspace ever. Here, this is not that clear. (follow-up is fine)